tools/analyzer_plugin/lib/src/assist/refs/add_create_ref.dart

```diff
@@ -26,31 +27,33 @@ typedef CreateRefLinkedEditFn = void Function(
 
 /// Utility function shared between diagnostics and assists to add a field to a
 /// component assigned to the return value of a `createRef()` call.
-void addCreateRef(
+void addUseOrCreateRef(
   DartFileEditBuilder builder,
   FluentComponentUsage usage,
   ResolvedUnitResult result, {
+  bool fromAssist = false,
   AnalyzerDebugHelper? debug,
 }) {
   const nameGroup = 'refName';
   const typeGroup = 'refType';
 
+  final isFnComponentUsage = getClosestFunctionComponent(usage.node) != null;
```

Comment:
#supernit This name could be mistaken for "usage of a function component".

Perhaps `isUsageInFnComponent`?